### PR TITLE
Fixes incorrect strategy init with HPUAccelerator

### LINKED ISSUE #19614: Initializing Trainer with accelerator=HPUAccelerator() and strategy="auto" initializes it with incorrect strategies
### Bug description

Initializing Trainer with accelerator=HPUAccelerator() (from lightning_habana) and strategy = "auto" initializes it with incorrect strategies. 
For single device, it initializes with SingleDeviceStrategy instead of SingleHPUStrategy,
For multiple devices, it inits with ddp instead of ParallelHPUStrategy.

Issue is not seen when setting accelerator="hpu".

### What version are you seeing the problem on?

v2.1, v2.2, master

### How to reproduce the bug

```python
from lightning import Trainer
from lightning.pytorch.demos.boring_classes import BoringModel
from lightning_habana import HPUAccelerator

model = BoringModel()

trainer = Trainer(
    accelerator=HPUAccelerator(),
    devices=1,
    strategy="auto",
    fast_dev_run=True,
)

trainer.fit(model)
```

### Error messages and logs

```
# Error message

Traceback (most recent call last):
  File "/home/agola/qnpu/qnplt/src/lightning-Habana/tests/test_pytorch/test_strategy.py", line 10, in <module>
    trainer = Trainer(
  File "/home/agola/qnpu/qnplt/src/pytorch-lightning/src/lightning/pytorch/utilities/argparse.py", line 70, in insert_env_defaults
    return fn(self, **kwargs)
  File "/home/agola/qnpu/qnplt/src/pytorch-lightning/src/lightning/pytorch/trainer/trainer.py", line 401, in __init__
    self._accelerator_connector = _AcceleratorConnector(
  File "/home/agola/qnpu/qnplt/src/pytorch-lightning/src/lightning/pytorch/trainer/connectors/accelerator_connector.py", line 161, in __init__
    self._lazy_init_strategy()
  File "/home/agola/qnpu/qnplt/src/pytorch-lightning/src/lightning/pytorch/trainer/connectors/accelerator_connector.py", line 587, in _lazy_init_strategy
    raise ValueError(
ValueError: The `HPUAccelerator` can only be used with a `SingleHPUStrategy` or `HPUParallelStrategy`, found SingleDeviceStrategy.
```

### Environment

<details>
  <summary>Current environment</summary>

* CUDA:
        - GPU:               None
        - available:         False
        - version:           None
* Lightning:
        - lightning:         2.3.0.dev0
        - lightning-cloud:   0.5.64
        - lightning-habana:  1.5.0.dev0
        - lightning-utilities: 0.10.1
        - pytorch-lightning: 2.2.1
        - torch:             2.2.0a0+git8964477
        - torch-tb-profiler: 0.4.0
        - torchaudio:        2.2.0+08901ad
        - torchdata:         0.7.1+5e6f7b7
        - torchmetrics:      0.11.4
        - torchtext:         0.17.0+400da5c
        - torchvision:       0.17.0+b2383d4
* System:
        - OS:                Linux
        - architecture:
                - 64bit
                - ELF
        - processor:         x86_64
        - python:            3.10.13
        - release:           5.15.0-91-generic
        - version:           #101-Ubuntu SMP Tue Nov 14 13:30:08 UTC 2023

</details>

### More info

_No response_

cc @jerome-habana

### PR DESCRIPTION
**Pull request by ankitgola005**
## What does this PR do?
Fixes for incorrect strategy init with HPUAccelerator 
Accelerator_connector only picks correct "auto" strategy when accelerator is set as "hpu". setting it with class instance fails this check and it sets default strategy from SingleDeviceStrategy / DDPStrategy instead of SingleHPUStrategy / ParallelHPUStrategy.

Fixes #19614 

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

----
📚 Documentation preview 📚: https://pytorch-lightning--19615.org.readthedocs.build/en/19615/

### CODE REVIEW

File: src/lightning/pytorch/trainer/connectors/accelerator_connector.py [hunk 1]
```diff
@@ -408,13 +408,16 @@ def _choose_and_init_cluster_environment(self) -> ClusterEnvironment:
         return LightningEnvironment()
 
     def _choose_strategy(self) -> Union[Strategy, str]:
```
Comment (awaelchli) [inline src/lightning/pytorch/trainer/connectors/accelerator_connector.py:410]: Can we refactor it into this logic? I think that would be a bit cleaner.

```py
if _habana_available_and_importable():
    from lightning_habana import HPUAccelerator

    if self._accelerator_flag == "hpu" or isinstance(self._accelerator_flag, HPUAccelerator):
        # select strategy

if self._accelerator_flag == "hpu" and not _habana_available_and_importable():
	raise ImportError(...)
```

Comment (ankitgola005) [reply src/lightning/pytorch/trainer/connectors/accelerator_connector.py:410]: yes, I was avoiding a top level import. This looks cleaner. Should also avoid the current pipeline fail.

File: src/lightning/pytorch/trainer/connectors/accelerator_connector.py [hunk 2]
```diff
@@ -408,21 +408,25 @@ def _choose_and_init_cluster_environment(self) -> ClusterEnvironment:
...
+                    return HPUParallelStrategy.strategy_name
+
+                from lightning_habana import SingleHPUStrategy
+
+                return SingleHPUStrategy(device=torch.device("hpu"))
+        if self._accelerator_flag == "hpu" and not _habana_available_and_importable():
+            raise ImportError(
+                "You have asked for HPU but you miss install related integration."
+                " Please run `pip install lightning-habana` or see for further instructions"
+                " in https://github.com/Lightning-AI/lightning-Habana/."
```
Comment (awaelchli) [inline src/lightning/pytorch/trainer/connectors/accelerator_connector.py:427]: While we are here, we can also fix the grammar in the text there:

```suggestion
                "You have asked for HPU but you miss a required dependency."
                " Please run `pip install lightning-habana` or see for further instructions"
                " in https://github.com/Lightning-AI/lightning-Habana/."
```

**Review by awaelchli — APPROVED** [review]
Thanks!

**Review by Borda — APPROVED** [review]

**Review by jerome-habana — APPROVED** [review]

**Pull request merged**